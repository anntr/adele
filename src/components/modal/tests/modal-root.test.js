import React from 'react';
import ModalRoot from '../modal-root';
import Modal from '../modal';

const mockFunc = jest.fn();
const MockChild = () => <div />;
const MockModalContent = () => <div />;

it('renders modal root correctly', () => {
  const wrapper = shallow(<ModalRoot closeModal={mockFunc} isVisible={false}><MockChild /></ModalRoot>);
  expect(wrapper).toMatchSnapshot();
});

it('renders modal when it should be visible', () => {
  const wrapper = shallow(<ModalRoot closeModal={mockFunc} isVisible component={MockModalContent}><MockChild /></ModalRoot>);
  expect(wrapper).toMatchSnapshot();
});