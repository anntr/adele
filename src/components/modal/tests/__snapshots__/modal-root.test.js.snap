// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders modal root correctly 1`] = `
<withState(withHandlers(withContext(Component)))
  closeModal={[Function]}
  isVisible={false}
  setComponent={[Function]}
>
  <MockChild />
</withState(withHandlers(withContext(Component)))>
`;

exports[`renders modal when it should be visible 1`] = `
<withState(withHandlers(withContext(Component)))
  closeModal={[Function]}
  isVisible={true}
  setComponent={[Function]}
>
  <MockChild />
</withState(withHandlers(withContext(Component)))>
`;
